src/pandroid/app/src/main/java/com/panda3ds/pandroid/view/PandaGlRenderer.java

```diff
@@ -0,0 +1,116 @@
+package com.panda3ds.pandroid.view;
+
+import static android.opengl.GLES32.*;
+
+import android.content.res.Resources;
+import android.graphics.Rect;
+import android.opengl.GLSurfaceView;
+import android.util.Log;
+import com.panda3ds.pandroid.AlberDriver;
+import com.panda3ds.pandroid.utils.Constants;
+import com.panda3ds.pandroid.view.renderer.ConsoleRenderer;
+import com.panda3ds.pandroid.view.renderer.layout.ConsoleLayout;
+import com.panda3ds.pandroid.view.renderer.layout.DefaultScreenLayout;
+import javax.microedition.khronos.egl.EGLConfig;
+import javax.microedition.khronos.opengles.GL10;
+
+public class PandaGlRenderer implements GLSurfaceView.Renderer, ConsoleRenderer {
+	private final String romPath;
+	private ConsoleLayout displayLayout;
+	private int screenWidth, screenHeight;
+	private int screenTexture;
+	public int screenFbo;
+
+	PandaGlRenderer(String romPath) {
+		super();
+		this.romPath = romPath;
+
+		screenWidth = Resources.getSystem().getDisplayMetrics().widthPixels;
+		screenHeight = Resources.getSystem().getDisplayMetrics().heightPixels;
+		setLayout(new DefaultScreenLayout());
+	}
+
+	@Override
+	protected void finalize() throws Throwable {
+		if (screenTexture != 0) {
+			glDeleteTextures(1, new int[] {screenTexture}, 0);
+		}
+		if (screenFbo != 0) {
+			glDeleteFramebuffers(1, new int[] {screenFbo}, 0);
+		}
+		super.finalize();
+	}
+
+	public void onSurfaceCreated(GL10 unused, EGLConfig config) {
+		Log.i("pandroid", glGetString(GL_EXTENSIONS));
+		Log.w("pandroid", glGetString(GL_VERSION));
+
+		glClearColor(0.0f, 0.0f, 0.0f, 1.0f);
+		glClear(GL_COLOR_BUFFER_BIT);
+
+		int[] generateBuffer = new int[1];
+		glGenTextures(1, generateBuffer, 0);
+		screenTexture = generateBuffer[0];
+		glBindTexture(GL_TEXTURE_2D, screenTexture);
+		glTexStorage2D(GL_TEXTURE_2D, 1, GL_RGBA8, screenWidth, screenHeight);
+		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+		glBindTexture(GL_TEXTURE_2D, 0);
+
+		glGenFramebuffers(1, generateBuffer, 0);
+		screenFbo = generateBuffer[0];
+		glBindFramebuffer(GL_FRAMEBUFFER, screenFbo);
+		glFramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, GL_TEXTURE_2D, screenTexture, 0);
```

Comment:
This should check that the framebuffer is complete via `glGetFramebufferStatus` before unbinding, and print something to logcat if the fb is not complete.

Mobile GPUs are finicky, so if we find a driver+GPU combo that gets mad at the framebuffer config (eg the non-POT texture or the texture type) then it will be a lot easier to debug with proper error reporting.